### PR TITLE
Add example to ClipOval docs

### PR DESCRIPTION
**Pull request by loic-sharma**

### CODE REVIEW

File: packages/flutter/lib/src/widgets/basic.dart [hunk 1]
```diff
@@ -925,6 +925,17 @@ class ClipRRect extends SingleChildRenderObjectWidget {
 /// prevents its child from painting outside that oval, but the size and
 /// location of the clip oval can be customized using a custom [clipper].
 ///
+/// {@tool snippet}
+///
+/// This example clips an image of a cat using an oval.
```
[the remark this answers is not in the document]
Comment (nate-thegrate) [reply packages/flutter/lib/src/widgets/basic.dart:930]: There are some crazy outliers too:

https://github.com/flutter/flutter/blob/a775f8a0b8326b6b407d086bad61c550ab77bbc2/packages/flutter/lib/src/material/time_picker.dart#L2967-L2968

**Review by nate-thegrate — APPROVED** [review]
LGTM!

I feel inspired to oval-clip my favorite cat photos 🐱

File: packages/flutter/lib/src/widgets/basic.dart [hunk 2]
```diff
@@ -925,6 +925,17 @@ class ClipRRect extends SingleChildRenderObjectWidget {
...
+///
+/// ```dart
+/// ClipOval(
+///   child: Image.network('images/cat.png')
```
[the remark this answers is not in the document]
Comment (loic-sharma) [reply packages/flutter/lib/src/widgets/basic.dart:934]: Good catch! I meant to switch this to `.asset` too, will fix!

[the remark this answers is not in the document]
Comment (loic-sharma) [reply packages/flutter/lib/src/widgets/basic.dart]: Fixed!

File: packages/flutter/lib/src/widgets/basic.dart (hunk 1 above)
[the remark this answers is not in the document]
Comment (loic-sharma) [reply packages/flutter/lib/src/widgets/basic.dart:930]: Hm interesting. The [documentation on snippets](https://github.com/flutter/flutter/tree/master/dev/snippets#snippet-tool) uses an empty line, I suspect that's the preferred style. I'll do a cleanup pass on this file once my doc PRs land :)

**Comment by Piinks** [comment]
(PR triage): @loic-sharma is this ready to merge?

**Comment by loic-sharma** [comment]
@Piinks Yup, was keeping it open so that folks could leave feedback. I'll start merging these in.

**Pull request merged**